### PR TITLE
First micro-ROS app on Linux reviewed.

### PR DESCRIPTION
**Pull request by FranFin**

### CODE REVIEW

**Comment by pablogs9** [comment]
LGTM

File: _docs/tutorials/core/first_application_linux/index.md [hunk 1]
```diff
@@ -7,29 +7,22 @@ This tutorial teaches you how to create a first micro-ROS application on Linux f
 
 ## Installing ROS 2 and the micro-ROS build system
 
-First of all, let's take a Ubuntu 18.04 LTS computer and install **ROS 2 Dashing Diademata**:
+First of all, install **ROS 2 Dashing Diademata** on your Ubuntu 18.04 LTS computer.
+To do so from binaries, via Debian packages, follow the instructions detailed
+[here](https://index.ros.org/doc/ros2/Installation/Dashing/Linux-Install-Debians/) instead.
```
Comment (ralph-lange) [inline _docs/tutorials/core/first_application_linux/index.md:12]:
```suggestion
from the [ROS Index](https://index.ros.org/doc/ros2/Installation/Dashing/Linux-Install-Debians/) website.
```

File: _docs/tutorials/core/first_application_linux/index.md [hunk 2]
```diff
@@ -7,29 +7,22 @@ This tutorial teaches you how to create a first micro-ROS application on Linux f
 
 ## Installing ROS 2 and the micro-ROS build system
 
-First of all, let's take a Ubuntu 18.04 LTS computer and install **ROS 2 Dashing Diademata**:
+First of all, install **ROS 2 Dashing Diademata** on your Ubuntu 18.04 LTS computer.
+To do so from binaries, via Debian packages, follow the instructions detailed
```
Comment (ralph-lange) [inline _docs/tutorials/core/first_application_linux/index.md:11]:
```suggestion
To do so from binaries, via Debian packages, follow the instructions
```

File: _docs/tutorials/core/first_application_linux/index.md [hunk 3]
```diff
@@ -63,87 +56,87 @@ Once the build system is installed, let's create a firmware workspace that targe
...
+A thorough review of this file is illustrative of how to create a micro-ROS publisher or subscriber.
+
+Once the app has been created, the build step is in order.
+Notice that, with respect to the four-steps workflow delined above, we would expect a configuration step to happen
+before building the app. However, given that we are compiling micro-ROS in the host machine rather than in a board,
+the cross-compilation implemented by this configure step is not required in this case.
+We can then proceed to build the firmware and source the local installation:
+
+```bash
+# Build step
```
Comment (ralph-lange) [inline _docs/tutorials/core/first_application_linux/index.md:103]:
```suggestion
```
I propose to remove this comment given the explanation in previous paragraph.

File: _docs/tutorials/core/first_application_linux/index.md [hunk 4]
```diff
@@ -63,87 +56,87 @@ Once the build system is installed, let's create a firmware workspace that targe
...
 
-The micro-ROS app is ready to be built and connected to a micro-ROS-Agent to start talking with the rest of the ROS 2 world.
+The micro-ROS app is now ready to be connected to a micro-ROS agent to start talking with the rest of the ROS 2
+world.
 
-First of all, create a micro-ROS agent:
+To do that, let's first of all create a micro-ROS agent:
 
 ```bash
 # Download micro-ROS-Agent packages
```
Comment (ralph-lange) [inline _docs/tutorials/core/first_application_linux/index.md:116]:
```suggestion
```
Comment is explained in the previous paragraph already.

File: _docs/tutorials/core/first_application_linux/index.md [hunk 5]
```diff
@@ -153,7 +146,12 @@ source install/local_setup.bash
...
-And finally, let's check that everything is working in another command line. We are going to listen to ping topic to check whether the Ping Pong node is publishing its own pings:
+## Testing the micro-ROS app
+
+Now, we want to check that everything is working.
+
+Open another command line. We are going to listen to the `ping` topic
+with ROS 2 to check whether the micro-ROS Ping Pong node is correctly publishing the expected pings:
 
 ```bash
 source /opt/ros/$ROS_DISTRO/setup.bash
```
Comment (ralph-lange) [inline _docs/tutorials/core/first_application_linux/index.md:157]:
```suggestion
source /opt/ros/dashing/setup.bash
```

... to be consistent with similar changes above.

File: _docs/tutorials/core/first_application_linux/index.md [hunk 6]
```diff
@@ -187,7 +189,7 @@ source /opt/ros/$ROS_DISTRO/setup.bash
 ros2 topic echo /microROS/pong
 ```
 
-At this point, we know that our app is publishing pings. Let's check if it also answers to someone else pings in a new command line:
+And now, let's publish a `fake_ping` with ROS 2 from yet another command line:
 
 ```bash
 source /opt/ros/$ROS_DISTRO/setup.bash
```
Comment (ralph-lange) [inline _docs/tutorials/core/first_application_linux/index.md:195]:
```suggestion
source /opt/ros/dashing/setup.bash
```

File: _docs/tutorials/core/first_application_linux/index.md [hunk 7]
```diff
@@ -228,18 +233,20 @@ frame_id: fake_ping
...
-One of the advantages of having an Linux micro-ROS app is that you don't need to buy a bunch of hardware in order to test some multi-node micro-ROS apps. So, with the same micro-ROS agent of the last section, let's open four different command lines and run the following on each:
+One of the advantages of having an Linux micro-ROS app is that you don't need to buy a bunch of hardware in order to
+test some multi-node micro-ROS apps.
+So, with the same micro-ROS agent of the last section, let's open four different command lines and run the following on
+each:
 
 ```bash
 cd microros_ws
 
 source /opt/ros/$ROS_DISTRO/setup.bash
```
Comment (ralph-lange) [inline _docs/tutorials/core/first_application_linux/index.md:246]:
```suggestion
source /opt/ros/dashing/setup.bash
```

**Review by ralph-lange — APPROVED** [review]

**Pull request closed without merging**